src/prototypecomponentsnavigatorwindow.cpp

```diff
@@ -0,0 +1,62 @@
+#include "prototypecomponentsnavigatorwindow.h"
+#include "ui_prototypecomponentsnavigatorwindow.h"
+#include <QItemSelectionModel>
+#include <QAbstractItemModel>
+#include <QStandardItem>
+
+
+PrototypeComponentsNavigatorWindow::PrototypeComponentsNavigatorWindow(QWidget *parent) :
+    QWidget(parent),
+    ui(new Ui::PrototypeComponentsNavigatorWindow)
+{
+    ui->setupUi(this);
+
+
+    ui->listView->setIconSize(QSize(100,100));
+    ui->listView->setUniformItemSizes(true);
+    ui->listView->setSelectionMode(QAbstractItemView::MultiSelection);
+
+    connect(ui->pushButtonAddComponentsToPrototipe, &QPushButton::clicked, this, &PrototypeComponentsNavigatorWindow::newComponentRequested);
+    connect(ui->pushButtonLoadSelectedComponents, &QPushButton::clicked, this, &PrototypeComponentsNavigatorWindow::componentOpenButtonClicked);
+    connect(ui->pushButtonDeleteSelectedComponent, &QPushButton::clicked, this, &PrototypeComponentsNavigatorWindow::componentRemovButtonClicked);
+
+}
+
+PrototypeComponentsNavigatorWindow::~PrototypeComponentsNavigatorWindow()
+{
+    delete ui;
+}
+
+void PrototypeComponentsNavigatorWindow::setModel(QAbstractItemModel *model)
+{
+    if(model)
+    {
+        ui->listView->setModel(model);
+        connect(ui->listView->selectionModel(), SIGNAL(selectionChanged(QItemSelection, QItemSelection)), this, SLOT(handleSelectionChanged(QItemSelection, QItemSelection)));
+    }
+}
+
+void PrototypeComponentsNavigatorWindow::componentOpenButtonClicked()
+{
+    emit componentOpened("");
+}
+
+void PrototypeComponentsNavigatorWindow::componentRemovButtonClicked()
+{
+    emit componentRemoved("");
+}
+
+void PrototypeComponentsNavigatorWindow::handleSelectionChanged(const QItemSelection &selected, const QItemSelection &deselected)
+{
+    for (const QModelIndex &index : selected.indexes())
```

Comment:
I do not understand what are you trying to do here? Why you need cycles in method `handleSelectionChanged`?